webapp/packages/core/src/blocks/InFocus.tsx

```diff
@@ -0,0 +1,31 @@
+import React, {
+  useEffect, useRef, ReactElement, MutableRefObject
+} from 'react';
+
+type InFocusProps = {
+  children: ReactElement;
+}
+
+export function InFocus({ children }: InFocusProps) {
+  const isFirstRender: MutableRefObject<boolean> = useRef(true);
+  const childRef: MutableRefObject<HTMLElement | null> = useRef(null);
+
+  useEffect(() => {
+    if (isFirstRender.current) {
+      isFirstRender.current = false;
+      if (childRef.current !== null) {
+        const firstFocusable: HTMLElement | null = childRef.current
+          .querySelector('button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])');
+        if (firstFocusable) {
+          firstFocusable.focus();
+        }
+
+      }
+    }
+  });
+  const childElement = React.Children.only(children);
+  return React.cloneElement(
+    childElement,
+    { ref: (el: HTMLElement) => childRef.current = el }
+  );
```

Comment:
first child can be not HTML element, so maybe better to wrap `<div {...props} ref={childRef}>{children}</div>`

Reply:
just don't use InFocus around non-html components
The Idea is to have inFocus as HOC the dosn't append something to DOM

Reply:
yep I understand that idea, in this case, may be better to replace InFocus with hook that returns ref, that we can pass to HTML element